scipio/src/file_stream.rs

```diff
@@ -0,0 +1,1338 @@
+    pub fn skip(&mut self, bytes: u64) {
+        let mut state = self.state.borrow_mut();
+        let buffer_id = state.buffer_id(self.current_pos);
+        self.current_pos = std::cmp::min(self.current_pos + bytes, self.end);
+        let new_buffer_id = state.buffer_id(self.current_pos);
+        let candidate_read_ahead_pos = align_down(self.current_pos, self.buffer_size);
+        state.buffer_read_ahead_current_pos = std::cmp::max(
+            state.buffer_read_ahead_current_pos,
+            candidate_read_ahead_pos,
+        );
+
+        // remember the end range is exclusive so if we didn't cross a buffer
+        // we don't discard anything
+        for id in buffer_id..new_buffer_id {
+            state.discard_buffer(id);
+        }
+        state.replenish_read_ahead(self.state.clone(), self.file.clone());
+    }
+
+    /// Acquires the current position of this [`StreamReader`].
+    ///
+    /// # Examples
+    /// ```no_run
+    /// use scipio::{DmaFile, StreamReaderBuilder, LocalExecutor};
+    ///
+    /// let ex = LocalExecutor::make_default();
+    /// ex.run(async {
+    ///     let file = DmaFile::open("myfile.txt").await.unwrap();
+    ///     let mut reader = StreamReaderBuilder::new(file).build();
+    ///     assert_eq!(reader.current_pos(), 0);
+    ///     reader.skip(8);
+    ///     assert_eq!(reader.current_pos(), 8);
+    ///     reader.close().await.unwrap();
+    /// });
+    /// ```
+    ///
+    /// [`StreamReader`]: struct.StreamReader.html
+    pub fn current_pos(&self) -> u64 {
+        self.current_pos
+    }
+
+    /// Allows access to the buffer that holds the current position with no extra copy
+    ///
+    /// In order to use this API, one must guarantee that reading the specified length may cross
+    /// into a different buffer.  Users of this API are expected to be aware of their buffer size
+    /// (selectable in the [`StreamReaderBuilder`]) and act accordingly.
+    ///
+    /// The buffer is also not released until the returned [`ReadResult`] goes out of scope. So
+    /// if you plan to keep this alive for a long time this is probably the wrong API.
+    ///
+    /// Let's say you want to open a file and check if its header is sane: this is a good API for
+    /// that.
+    ///
+    /// But if after such header there is an index that you want to keep in memory, then you
+    /// are probably better off with one of the methods from [`AsyncReadExt`].
+    ///
+    /// # Examples
+    /// ```no_run
+    /// use scipio::{DmaFile, StreamReaderBuilder, LocalExecutor};
+    ///
+    /// let ex = LocalExecutor::make_default();
+    /// ex.run(async {
+    ///     let file = DmaFile::open("myfile.txt").await.unwrap();
+    ///     let mut reader = StreamReaderBuilder::new(file).build();
+    ///     assert_eq!(reader.current_pos(), 0);
+    ///     reader.skip(8);
+    ///     assert_eq!(reader.current_pos(), 8);
+    ///     reader.close().await.unwrap();
+    /// });
```

Comment:
This example does not demonstrate how to use `get_buffer_aligned `

Reply:
I will add a commit fixing this